luigi/contrib/k8s_job.py

```diff
@@ -0,0 +1,193 @@
+        self.job_uuid = str(uuid.uuid4().hex)
+        self.uu_name = self.name + "-luigi-" + self.job_uuid
+        if ("restartPolicy" not in self.spec_schema):
+            self.spec_schema["restartPolicy"] = "Never"
+
+    @property
+    def auth_method(self):
+        """
+        This can be set to ``kubeconfing`` or ``ServiceAccount``.
+        It defaults to ``kubeconfing``.
+
+        For more details please referer to:
+
+        - kubeconfing: http://kubernetes.io/docs/user-guide/kubeconfig-file
+        - ServiceAccount: http://kubernetes.io/docs/user-guide/service-accounts
+        """
+        return configuration.get_config().get("k8s", "auth_method", "kubeconfing")
+
+    @property
+    def kubeconfig_path(self):
+        """
+        Path to kubeconfing file, for cluster authentication.
+        It defaults to "~/.kube/config", which is the default location
+        when using minikube (http://kubernetes.io/docs/getting-started-guides/minikube).
+        When auth_method is ``ServiceAccount`` this properity is ignored.
+
+        **WARNING**: For Python versions < 3.5 kubeconfing must point to a Kubernetes API
+        hostname, and NOT to an IP address.
+
+        For more details please referer to:
+        http://kubernetes.io/docs/user-guide/kubeconfig-file
+        """
+        return configuration.get_config().get("k8s", "kubeconfig_path", "~/.kube/config")
+
+    @property
+    def name(self):
+        """
+        A name for this job. This task will automatically append a UUID to the
+        name before to submit to Kubernetes.
+        """
+        pass
+
+    @property
+    def spec_schema(self):
+        """
+        Kubernetes Job spec schema in JSON format, example::
+
+            {
+                "containers": [{
+                    "name": "pi",
+                    "image": "perl",
+                    "command": ["perl",  "-Mbignum=bpi", "-wle", "print bpi(2000)"]
+                }],
+                "restartPolicy": "Never"
+            }
+
+        **restartPolicy**
+
+        - If restartPolicy is not defined, it will be set to "Never" by default.
+
+        - **Warning**: restartPolicy=OnFailure will bypass max_retrials, and restart
+        the container until success, with the risk of blocking the Luigi task.
+
+        For more informations please refer to:
+        http://kubernetes.io/docs/user-guide/pods/multi-container/#the-spec-schema
+        """
+
+    @property
+    def max_retrials(self):
+        """
+        Maximum number of retrials in case of failure.
+        """
+        return 0
+
+    def __track_job(self):
+        """Poll job status while active"""
+        while (self.__get_job_status() == "running"):
+            self.__logger.debug("Kubernetes job " + self.uu_name + " is still running")
+            time.sleep(self.__POLL_TIME)
+        if(self.__get_job_status() == "succeeded"):
+            self.__logger.info("Kubernetes job " + self.uu_name + " succeeded")
```

Comment:
I think it would make sense to touch some sort of output at this point to signal job completion.